quickwit-cli/src/split.rs

```diff
@@ -115,12 +187,113 @@ impl SplitCliCommand {
 
     pub async fn execute(self) -> anyhow::Result<()> {
         match self {
+            Self::List(args) => list_split_cli(args).await,
             Self::Describe(args) => describe_split_cli(args).await,
             Self::Extract(args) => extract_split_cli(args).await,
         }
     }
 }
 
+pub async fn list_split_cli(args: ListSplitArgs) -> anyhow::Result<()> {
```

Comment:
This function should be decomposed into at least 2 functions: one to perform the io and filtering, the second to display the results. That way, the former function can be tested. I'd personally go with 3 steps:
1. lists the splits
2. parse the args and filter the splits
3. display the output